From 5d60f6beb5ddcda0aee99261d0bb7fa7db934001 Mon Sep 17 00:00:00 2001
From: "Jason A. Donenfeld" <Jason@zx2c4.com>
Date: Mon, 18 Mar 2019 14:41:10 -0600
Subject: wg-quick: freebsd: export TMPDIR when restoring and don't make empty

Otherwise mktemp doesn't see it, and if it's empty we wind up in /.
---
 src/tools/wg-quick/freebsd.bash | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/tools/wg-quick/freebsd.bash b/src/tools/wg-quick/freebsd.bash
index e83dbef0..c3fa47a5 100755
--- wg-quick/freebsd.bash
+++ wg-quick/freebsd.bash
@@ -42,8 +42,9 @@ unset ORIGINAL_TMPDIR
 make_temp() {
 	local old_umask
 
-	[[ -v ORIGINAL_TMPDIR ]] && TMPDIR="$ORIGINAL_TMPDIR"
+	[[ -v ORIGINAL_TMPDIR ]] && export TMPDIR="$ORIGINAL_TMPDIR"
 	ORIGINAL_TMPDIR="$TMPDIR"
+	[[ -z $TMPDIR ]] && unset TMPDIR
 
 	old_umask="$(umask)"
 	umask 077
-- 
cgit v1.2.1-20-gc37e

